### PR TITLE
HW02: Implement my own Array methods(a_serikov)

### PR DESCRIPTION
**Pull request by Nij4life**

### CODE REVIEW

File: HW02/aleksandr_serikov/array_refinements.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+require 'pry'
```
Comment (B0RlS) [inline HW02/aleksandr_serikov/array_refinements.rb:1]:
```suggestion
```
You don't need it

File: HW02/aleksandr_serikov/array_refinements.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      for i in 0...size
+        yield self[i] 
+      end
+      self
+    end
+    
+    def map
+      new_array = []
+      for i in 0...size
+        new_array << (self[i] && yield(self[i]))
```
Comment (vladhilko) [inline HW02/aleksandr_serikov/array_refinements.rb:17]: Why do you need it? `1 && 2` => `2`
```suggestion
        new_array <<  yield(self[i])
```

File: HW02/aleksandr_serikov/array_refinements.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      end
+    
+    new_array
+    end
```
Comment (vladhilko) [inline HW02/aleksandr_serikov/array_refinements.rb:21]:
```suggestion
      new_array
    end
```

**Review by B0RlS — APPROVED** [review]

File: HW02/aleksandr_serikov/array_refinements.rb (hunk 1 above)
Comment (vladhilko) [inline HW02/aleksandr_serikov/array_refinements.rb:1]:
```suggestion

```

**Review by vladhilko — APPROVED** [review]

File: HW02/aleksandr_serikov/array_refinements.rb (hunk 1 above)
Comment (Nij4life) [reply HW02/aleksandr_serikov/array_refinements.rb:1]: Done

File: HW02/aleksandr_serikov/array_refinements.rb (hunk 2 above)
Comment (Nij4life) [reply HW02/aleksandr_serikov/array_refinements.rb:17]: Done

File: HW02/aleksandr_serikov/array_refinements.rb (hunk 3 above)
Comment (Nij4life) [reply HW02/aleksandr_serikov/array_refinements.rb:21]: Done

**Pull request merged**